src/perftest_resources.c

```diff
@@ -1566,23 +1566,41 @@ int destroy_ctx(struct pingpong_context *ctx,
  *
  ******************************************************************************/
 #ifdef HAVE_EX_ODP
-static int check_odp_support(struct pingpong_context *ctx)
+static int check_odp_support(struct pingpong_context *ctx, VerbType verb)
 {
 	struct ibv_device_attr_ex dattr;
-	int odp_support_send = IBV_ODP_SUPPORT_SEND;
-	int odp_support_recv = IBV_ODP_SUPPORT_RECV;
 	int ret = ibv_query_device_ex(ctx->context, NULL, &dattr);
 
 	if (ret) {
-		fprintf(stderr, " Couldn't query device for on-demand paging capabilities.\n");
+		fprintf(stderr, " Couldn't query device for On-Demand Paging capabilities.\n");
 		return 0;
-	} else if (!(dattr.odp_caps.per_transport_caps.rc_odp_caps & odp_support_send)) {
-		fprintf(stderr, " Send is not supported for RC transport.\n");
-		return 0;
-	} else if (!(dattr.odp_caps.per_transport_caps.rc_odp_caps & odp_support_recv)) {
-		fprintf(stderr, " Receive is not supported for RC transport.\n");
+	}
+
+	/* These capabilities must be set by device drivers. */
+	if (!(dattr.odp_caps.general_caps & IBV_ODP_SUPPORT)) {
+		fprintf(stderr, " The device does not support On-Demand Paging.\n");
 		return 0;
 	}
+
+	if (verb == SEND && (
```

Comment:
Thanks ! ,Great way of checking!, but we miss here reading the capabilities for other transports (DC, UD ,etc..), can you please add them?